airbyte-integrations/connectors/source-snipeit/source_snipeit/full_refresh_streams.py

```diff
@@ -0,0 +1,196 @@
+        self.limit_per_page: int = 500
+        self.total: int = 0
+        self.offset: int = 0
+        self.config: dict = config
+
+        # NOTE: This is probably not the best idea.
+        self.stop_immediately = False
+
+    @property
+    def url_base(self):
+        return self.config["base_url"]
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        """
+        This method should return a Mapping (e.g: dict) containing whatever information required to make paginated requests. This dict is passed
+        to most other methods in this class to help you form headers, request bodies, query params, etc..
+
+        For example, if the API accepts a 'page' parameter to determine which page of the result to return, and a response from the API contains a
+        'page' number, then this method should probably return a dict {'page': response.json()['page'] + 1} to increment the page count by 1.
+        The request_params method should then read the input next_page_token and set the 'page' param to next_page_token['page'].
+
+        :param response: the most recent response from the API
+        :return If there is another page in the result, a mapping (e.g: dict) containing information needed to query the next page in the response.
+                If there are no more pages in the result, return None.
+        """
+        if self.stop_immediately:
+            return {}
+        elif self.offset < self.total:
+            self.offset += self.limit_per_page
+            return {"offset": self.offset}
+        else:
+            return {}
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        """
+        Usually contains common params e.g. pagination size etc.
+        """
+        if next_page_token:
+            return {'limit': self.limit_per_page, 'offset': next_page_token.get("offset", None)}
+        else:
+            return {'limit': self.limit_per_page, 'offset': self.offset}
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        self.total = response.json().get("total", 0)
+        yield from response.json().get("rows", [])
+
+class Hardware(SnipeitStream):
+    primary_key = "id"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "hardware/"
+
+class Companies(SnipeitStream):
+    primary_key = "id"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "companies/"
```

Comment:
Can you run format on this files?

Reply:
What do you prefer for formatting? `black`? or something else?

Reply:
run `./gradlew airbyte-integrations:connectors:source-snipeit:blackFormat`